cmd/launcher/updater-finalizer.go

```diff
@@ -0,0 +1,34 @@
+// +build !windows
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+)
+
+// launcherFinalizer finalizes a launcher update. It assume the new
+// binary has been copied into place, and calls exec, so we start a
+// new running launcher in our place.
+func launcherFinalizer(logger log.Logger, shutdownOsquery func() error) func() error {
```

Comment:
`updateFinalizer` or `launcherUpdateFinalizer` would be a better name for this perhaps.

Reply:
reasonable